src/common/helpers/schema.helper.ts

```diff
@@ -6,6 +6,54 @@ import {
   NON_BF_GROUP_TYPE,
 } from '@common/constants/bibframeMapping.constants';
 import { PREV_ENTRY_PATH_INDEX } from '@common/constants/bibframe.constants';
+import { RecordNormalizingService } from '@common/services/recordNormalizing';
+import { RecordToSchemaMappingService } from '@common/services/recordToSchemaMapping';
+
+type IGetSchemaAndUserValuesFromRecord = {
+  base: Map<string, SchemaEntry>;
+  record: RecordEntry;
+  block?: string;
+  reference?: RecordReference;
+  recordBlocks: RecordBlocksList;
+  selectedEntriesService: ISelectedEntries;
+  schemaWithDuplicatesService: ISchemaWithDuplicates;
+  userValuesService: IUserValues;
+  commonStatusService: ICommonStatus;
+  template?: ResourceTemplateMetadata[];
+};
+
+export const getSchemaAndUserValuesFromRecord = async ({
```

Comment:
Not sure if the helper is the best place to do this.
We have `ServicesContext` and custom hooks for this purpose.
Please add a comment that this code should be refactored.